addons/web/static/tests/views/kanban_tests.js

```
@@ -839,6 +839,73 @@ QUnit.module('Views', {
         kanban.destroy();
     });
 
+    QUnit.test('Do not open record when clicking on `a` with `href`', function (assert) {
+        assert.expect(6);
+
+        delete this.data.partner.fields.state;
+        this.data.partner.records = [
+            { id: 1, foo: 'yop' },
+        ];
+
+        var isRecordOpen = false;
+
+        var kanban = createView({
+            View: KanbanView,
+            model: 'partner',
+            data: this.data,
+            arch: '<kanban class="o_kanban_test">' +
+                        '<templates>' +
+                            '<t t-name="kanban-box">' +
+                                '<div class="oe_kanban_global_click">' +
+                                    '<field name="foo"/>' +
+                                    '<div>' +
+                                        '<a class="o_test_link" href="#">test link</a>' +
+                                    '</div>' +
+                                '</div>' +
+                            '</t>' +
+                        '</templates>' +
+                    '</kanban>',
+            intercepts: {
+                // when clicking on a record in kanban view,
+                // it switches to form view.
+                switch_view: function () {
+                    isRecordOpen = true;
```

now that you have other asserts, i'd throw an exception here